apps/src/templates/manageStudents/ManageStudentsTable.jsx

```diff
@@ -0,0 +1,284 @@
+import React, {Component, PropTypes} from 'react';
+import color from "../../util/color";
+import {Table, sort} from 'reactabular';
+import wrappedSortable from '../tables/wrapped_sortable';
+import orderBy from 'lodash/orderBy';
+import PasswordReset from './PasswordReset';
+import ShowSecret from './ShowSecret';
+import i18n from "@cdo/locale";
+
+export const studentSectionDataPropType = PropTypes.shape({
+  id: PropTypes.number,
+  name: PropTypes.string,
+  username: PropTypes.string,
+  userType: PropTypes.string,
+  age: PropTypes.number,
+  gender: PropTypes.string,
+  secretWords: PropTypes.string,
+  secretPictureName: PropTypes.string,
+  secretPicturePath: PropTypes.string,
+  sectionId: PropTypes.number,
+});
+
+const GENDERS = {
+  'm': i18n.genderMale(),
+  'f': i18n.genderFemale()
+};
+
+/** @enum {number} */
+export const COLUMNS = {
+  NAME: 0,
+  AGE: 1,
+  GENDER: 2,
+  PASSWORD: 3,
+  ACTIONS: 4,
+};
+
+export const styles = {
+  table: {
+    width: 970,
+    borderRadius: 5,
+    color: color.charcoal,
+    backgroundColor: color.table_light_row
+  },
+  cell: {
+    border: '1px solid',
+    borderColor: color.border_light_gray,
+    padding: 10,
+    fontSize: 14,
+  },
+  headerCell: {
+    border: '1px solid',
+    borderColor: color.border_light_gray,
+    padding: 20,
+    backgroundColor: color.lightest_gray,
+    color: color.charcoal
+  },
+  cellFirst: {
+    borderWidth: '1px 0px 1px 1px',
+    borderColor: color.border_light_gray,
+    padding: 15,
+    width: 270
+  },
+  headerCellFirst: {
+    borderWidth: '1px 0px 1px 1px',
+    borderColor: color.border_light_gray,
+    padding: 15
+  },
+  link: {
+    color: color.teal,
+    fontFamily: '"Gotham 5r", sans-serif',
+    fontSize: 14,
+    textDecoration: 'none'
+  },
+};
+
+// Cell formatters.
+const nameFormatter = function (name, {rowData}) {
+  const url = '/teacher-dashboard#/sections/' + rowData.sectionId + '/student/' + rowData.id;
+  return (<div>
+    <a style={styles.link} href={url} target="_blank">{name}</a>
+    {rowData.loginType === 'email' &&
+      <p>{i18n.usernameLabel() + rowData.username}</p>
+    }
+  </div>);
+};
+
+const ageFormatter = function (age, {rowData}) {
```

Comment:
nit: For all of these, it should probably either be `function ageFormatter (age, {rowData}) {` or `const ageFormatter = (age, {rowData}) => {`